--- CMakeLists.txt	2023-04-24 18:28:13.303361854 +0200
+++ ../../BUILD/CMakeLists.txt	2023-04-24 15:35:19.480556313 +0200
@@ -13,10 +13,10 @@
 # This code is released under the libpng license. For conditions of distribution
 # and use, see the disclaimer and license in png.h
 
-cmake_minimum_required(VERSION 3.1)
-cmake_policy(VERSION 3.1)
+cmake_minimum_required(VERSION 3.16)
+cmake_policy(VERSION 3.16)
 
-project(libpng C ASM)
+project(libpng C ASM CXX)
 enable_testing()
 
 set(PNGLIB_MAJOR 1)
@@ -30,6 +30,10 @@
 set(PNGLIB_SHARED_VERSION
     ${PNGLIB_SHARED_SOVERSION}.${PNGLIB_REVISION}.${PNGLIB_SUBREVISION})
 
+# Enable fuzz testing with cifuzz
+find_package(cifuzz NO_SYSTEM_ENVIRONMENT_PATH)
+enable_fuzz_testing()
+
 include(GNUInstallDirs)
 
 # Allow users to specify location of zlib. Useful if zlib is being built
@@ -1182,5 +1186,11 @@
     FILE lib${PNG_LIB_NAME}.cmake)
 endif()
 
+add_fuzz_test(my_fuzz_test_1 my_fuzz_test_1.cpp)
+# add_fuzz_test(my_fuzz_test_1 my_fuzz_test_1.cpp png_mutator.cpp)
+target_link_libraries(my_fuzz_test_1 PRIVATE ${ZLIB_LIBRARY} ${M_LIBRARY} png16)
+target_include_directories(my_fuzz_test_1 PRIVATE .)
+target_compile_options(my_fuzz_test_1 PRIVATE)
+
 # TODO: Create MSVC import lib for MinGW-compiled shared lib. pexports
 # libpng.dll > libpng.def lib /def:libpng.def /machine:x86
